cardshifter-core/src/main/java/com/cardshifter/core/groovy/SentenceBuilder.groovy

```diff
@@ -0,0 +1,71 @@
+package com.cardshifter.core.groovy
+
+/**
+ * Build sentences using a convenient DSL. Sentences are capitalized and end with a period.
+ */
+class SentenceBuilder {
+
+    private StringBuilder stringBuilder = new StringBuilder()
+
+    static String build(@DelegatesTo(strategy = Closure.DELEGATE_FIRST, value = SentenceBuilder) Closure closure) {
+        def builder = new SentenceBuilder()
+        closure.delegate = builder
+        closure.resolveStrategy = Closure.DELEGATE_FIRST
+        closure.call()
+
+        return builder.toString()
+    }
+
+    String toString() {
+        String sentence = stringBuilder.toString()
+
+        sentence = sentence.trim()
+        if (sentence.length() > 0 && !sentence.endsWith('.')) {
+            sentence = sentence + '.'
+        }
+        sentence = sentence.capitalize()
+
+        return sentence
+    }
+
+    /**
+     * Add some text.
+     */
+    void text(String string) {
+        stringBuilder.append(string)
+    }
+
+    /**
+     * Add a separator if not already present and it's not the start of the sentence.
+     * @param sep The separator, e.g. ' ' or ', '.
+     */
+    void separator(String sep) {
+        if (stringBuilder.length() > 0 && !stringBuilder.toString().endsWith(sep)) {
+            stringBuilder.append(sep)
+        }
+    }
+
+    /**
+     * Add a list like 'egg, bacon and ham'.
+     * @param conjunction Conjunction between the last two elements, e.g. 'and'.
+     * @param elements A list of the elements, e.g. ['egg', 'bacon', 'ham'].
+     */
+    void list(String conjunction, List<String> elements) {
```

Comment:
I have a feeling this method can be refactored slightly. The indentation is a bit deep and maybe the logic can be approved. Just a feeling I have at the moment though. Maybe something for a certain Code Review site?